### DISCUSSION TITLE
Unable to compile Android 12 following wiki

### DISCUSSION THREAD

**Discussion by Solace50** [phhusson/treble_experimentations#2619] [Q&A]
I'm currently following the wiki page,

https://github.com/phhusson/treble_experimentations/wiki/How-to-build-a-GSI%3F

Which gets to step,

#4
Modify the source to fix issues in other devices using [phh patches](https://github.com/phhusson/treble_experimentations/releases/latest/download/patches-for-developers.zip):

    Download [the patches](https://github.com/phhusson/treble_experimentations/releases/latest/download/patches-for-developers.zip), then apply each patch in its project:

git am patch

Which is suppose to be ran on each file from the compressed download? Other sites list patch -p1 < patch which also fails. Is the working directory for this within where you run repo sync?

#5 
Go to the phh device repo and run bash generate.sh vendor/lineage/config/common_full_phone.mk - adapting for the ROM you wish to build for.

This does not exist?

#6
Lunch the [build variant](https://github.com/phhusson/treble_experimentations/blob/master/build.sh#L380) you want (ex. treble_arm64_avN-userdebug) and start the build

.build/envsetup.sh
lunch treble_arm64_avN-userdebug
WITHOUT_CHECK_API=true make -j$(nproc) systemimage

Does not exist when following the wiki guide... Is there a list of arguments to pass here or syntax for things like A only/Vanilla/non su etc...?

This is compiling for android 12

**Comment by User8395** [comment]
@phhusson?
